File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int n = nums.size();
        if (n == 1) {
            return true;
        }
        int count = 0,i;
        for (i = 1; i < n; i++) {
            if (nums[i] < nums[i - 1]) {
                count++;
            }
        }
        if (nums[n-1] > nums[0]) {
            count++;
        }
        return count<=1;
    }
};

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int n = nums.length;
        if(n == 1) {
            return true;
        }
        int count = 0;
        for(int i = 1; i<n; i++) {
            if(nums[i] < nums[i-1]) {
                count++;
            }
        }
        if(nums[n-1] > nums[0]) 
            count++;
        return count<=1; 
    }
}